PiwikTracker/Queue.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  EventQueue.swift
+//  PiwikTracker
+//
+//  Created by Cornelius Horstmann on 09.10.16.
+//  Copyright © 2016 Mattias Levin. All rights reserved.
+//
+
+import Foundation
+
+/// An Element that is queueable.
+protocol Queueable {
+    /// A unique identifier. It should be unique per object.
+    var id: UUID { get }
+    
+    /// This property is used to get the data that should be stored in the queue.
+    var data: Data { get }
+    
+    /// This initializer is used when dequeueing.
+    init?(data: Data)
```

Comment:
Not sure this will work. 
The queue implementation will not/should not know the original type is it sent, only that it can call `data`. It can only return objects of `Queueable` and the protocol should not care of how it is instantiated.